File: Sheep/Sheep.cpp
#include "Sheep.h"

Sheep::Sheep(World *world, int x, int y, int parent1Id, int parent2Id) :
    Animal(SHEEP_STRENGTH,
           SHEEP_INITIATIVE,
           SHEEP_SYMBOL,
           SHEEP_COLOR,
           world,
           x,
           y,
           parent1Id,
           parent2Id) {}

void Sheep::action() {
  Animal::action();
}

void Sheep::collision(Organism *organism) {
  Animal::collision(organism);
}

Organism* Sheep::createChild(int x, int y, int parent1Id, int parent2Id) {
  return new Sheep(this->getWorld(), x, y, parent1Id, parent2Id);
}

std::string Sheep::getType() const { return "Sheep"; }

File: src/Sheep/Sheep.java
package Sheep;

import Animal.Animal;
import Field.Field;
import Organism.Organism;
import Organism.OrganismData;
import Settings.Settings;
import World.World;

public class Sheep extends Animal {

    public Sheep(World world, Field field, int parent1Id, int parent2Id) {
        super(new OrganismData(world.getNextOrganismId(), Settings.SHEEP_INITIATIVE, Settings.SHEEP_STRENGTH, Settings.SHEEP_IMAGE, parent1Id, parent2Id), world, field);
    }

    @Override
    public String getType() {
        return "Sheep";
    }

    @Override
    public void reproduce(Field field, int parent1Id, int parent2Id) {
        Organism sheep = new Sheep(this.world, field, parent1Id, parent2Id);
        this.world.addOrganism(sheep);
    }

}
